src/sinks/splunk_hec.rs

```diff
@@ -721,15 +721,17 @@ mod integration_tests {
             .build()
             .unwrap();
 
-        // Wait for port 8089 to be reachable before firing off request
-        wait_for_tcp_duration("127.0.0.1:8089".parse().unwrap(), Duration::from_secs(30)).await;
-
-        let res = client
-            .get("https://localhost:8089/services/data/inputs/http?output_mode=json")
-            .basic_auth(USERNAME, Some(PASSWORD))
-            .send()
-            .await
-            .unwrap();
+        let res = retry_until(
+            || {
+                client
+                    .get("https://localhost:8089/services/data/inputs/http?output_mode=json")
+                    .basic_auth(USERNAME, Some(PASSWORD))
+                    .send()
```

Comment:
Does this `.send` call fail if the remote returns status code that's `>= 400`? We might want to add an accompanying wrapper to treat those as errors for the purposes of this functionality to reduce the possibility of race even further - as there are some services that start listening on the socket and responding to the HTTP requests before properly initializing the internal routes - and thus responding with 404 in a first few moments before after boot.

Reply:
According to [`send` docs](https://docs.rs/reqwest/0.9.21/reqwest/struct.RequestBuilder.html#method.send):

> This method fails if there was an error while sending request, redirect loop was detected or redirect limit was exhausted.

There's [.error_for_status](https://docs.rs/reqwest/0.10.8/src/reqwest/async_impl/response.rs.html#337-343), which turns a 400+ response into an error.

Reply:
Great, I'm not familiar with `reqwest`, but it should've guessed they have a built-in helper for that :smile: 
Shall we use in this case?

Reply:
I'll hold off for now, to scope this PR to just the retry logic, but I think it's worth doing a pass for HTTP tests in general, to ensure we're handling `404`s, etc, as you mentioned.